python/lsst/analysis/drp/calcFunctors.py

```diff
@@ -71,6 +72,43 @@ def __call__(self, df):
         return magDiff
 
 
+class ExtinctionCorrectedMagDiff(DataFrameAction):
+    magDiff = ConfigurableActionField(doc="Action that returns a difference in magnitudes",
+                                          default=MagDiff, dtype=DataFrameAction)
+    ebv = Field(doc="E(B-V) Column", dtype=str, default="ebv")
+    # These default extinction coefficients are WRONG, but better than nothing
+    # TO DO DM-34061: They should be computed from the FGCM lookup-tables
+    # Which store the standard bandpasses in a calib-like object
+    extinctionCoeffs = DictField(
+        doc="Dictionary of extinction coefficients for conversion from E(B-V) to extinction, A_filter",
+        keytype=str, itemtype=float,
+        default={"u": 4.505,
+                 "g": 3.240,
+                 "r": 2.276,
+                 "i": 1.633,
+                 "z": 1.263,
+                 "y": 1.075
+                 }
+    )
+
+    @property
+    def columns(self):
+        return self.magDiff.columns + (self.ebv,)
+
+    def __call__(self, df):
+        diff = self.magDiff(df)
+        ebv = df[self.ebv].values
+        col1Band = self.magDiff.col1.split('_')[0]
+        col2Band = self.magDiff.col2.split('_')[0]
+        av1 = self.extinctionCoeffs[col1Band]
```

Comment:
It might be better with a useful error message here in case someone tries to run it with a filter name not in the dictionary.

Reply:
Yes, either a more helpful error or (alternatively) log a warning as above and don't apply extinction corrections. (And below for `col2Band`).